Add FromIterator impl for FeatureCollection

rmanoka opened the pull request:
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

## Implementation Checklist

To summarize, this is the impl. checklist I'm suggesting:

1. [x] `FromIterator for FeatureCollection` 
2. [ ] ~~`From<&'a T> for Feature where From<&'a T> for Value`, T: BoundingRect~~
3. [x] Ensure correct bbox calculations in 1 ~~and 2 above~~


@michaelkirk @urschrei I realize we can't use `BoundingRect` as we only depend on `geo-types`.  Now, I'm a bit hesitant to add impl from geo-types -> `Feature` given it's just a conversion to `Value`, then to `Feature`.  The users could do this themselves by converting to `Value` then using `.into()` method. 

The PR currently just adds `FromIterator` impl.  Should add some tests for this;  have an examples collections with bbox to try out with?  We could also improve the docs for `Value`; currently it is hidden deep, and doesn't explain usage.

michaelkirk commented:
>  I realize we can't use BoundingRect as we only depend on geo-types

Oh right... shoot.

> The PR currently just adds FromIterator impl.

The FromIterator implementation looks great to me.

> have an examples collections with bbox to try out with

I don't have anything handy.

rmanoka commented:
Added some minimal test and minor updates to the docs.  The main page doc already quite nicely explains how to convert from geo-types, so I just added an example in the `FeatureCollection` page.

Addresses #170 to some extent

michaelkirk commented on src/feature_collection.rs:
````diff
@@ -47,6 +48,21 @@ use crate::{util, Bbox, Feature};
+///
+/// ```rust
+/// # extern crate geojson;
+/// # fn main() {
````
nbd: this `main` declaration can go away, right?

https://doc.rust-lang.org/rustdoc/documentation-tests.html#pre-processing-examples

(thanks for writing an example doc though!)

rmanoka replied:
I just copied the previous example.  I guess it's for the `extern crate` line above `fn main()`.

rmanoka replied:
Seems to work without the `/// # ` lines, so have removed them from both examples.

michaelkirk commented on src/feature_collection.rs:
```diff
@@ -144,3 +160,108 @@ impl<'de> Deserialize<'de> for FeatureCollection {
+        assert!(fc.bbox.is_some());
+        let bbox = fc.bbox.as_ref().unwrap();
+        assert_eq!(bbox.len(), 6);
+        for (i, coord) in bbox.iter().enumerate() {
```
To make this test more declarative, could we equally write something like:

`assert_eq!(bbox, vec![-1., -1., -1, 11., 11., 11.])`

rmanoka replied:
Well, I wanted to avoid any exact comparisons with float values.

rmanoka replied:
Your suggestion also works, and it does look neater.  I guess with bbox, it's just min/max, so we should be fine even if we change the coords in the future.

michaelkirk reviewed (approved):
LGTM!

All my feedback here should be considered optional - feel free to merge when you're ready.

rmanoka commented:
I'll leave this open for a day or two if @urschrei would like to add anything.  Otherwise, will merge and close #170 .

urschrei commented:
Very happy for this to merge!

rmanoka commented:
bors r=michaelkirk

lnicola commented:
bors r=michaelkirk

michaelkirk commented:
bors r+

(I've also added @rmanoka and @lnicola to be bors reviewers)
